commit be45aa83ccc16a570064d2512bd807a1b01b6dbd
Author: GeorgeClark <GeorgeClark@0b4bb1d4-4e5a-0410-9cc4-b2b747904278>
Date:   Mon Nov 26 04:32:23 2012 +0000

    Item11267: Don't reinstall extensions
    
    Some existing extensions have dependencies based on SVN Rev strings.  New
    modules as of 1.1.6, don't have the rev string available.
    
    Dependency.pm is changed to assume that if a dependency is trying to
    compare a svn revision to an installed dotted decimal version, then the
    dependency is automatically met.
    
    DEPENDENCY:  This patch must be applied AFTER Item11267-001.patch
    
    git-svn-id: http://svn.foswiki.org/branches/Release01x01@16103 0b4bb1d4-4e5a-0410-9cc4-b2b747904278

~~~PATCH 8da49fcc5b7babb0661a1ec2979a03c7:17867f295278e3bd355bcce0327a60e3 lib/Foswiki/Configure/Dependency.pm (Foswiki 1.1.5)
--- lib/Foswiki/Configure/Dependency.pm     2012-11-15 11:13:21.346451222 -0500
+++ lib/Foswiki/Configure/Dependency.pm.new 2012-11-28 11:35:17.931786783 -0500
@@ -459,6 +459,7 @@
         #print STDERR "Expecting SVN comparison, but RELEASE was $baseType \n";
         ( $baseType, @atuple ) = _decodeReleaseString($aVERSION)
           if ( defined $aVERSION && $baseType ne 'svn' );
+        return 1 if ( $baseType eq 'tuple' );
         return 0 unless ( $baseType eq 'svn' );
 
     }

~~~PATCH 221ccfdb31d41b8d376a5e095003e88d:70a6a0fd9f6a377f212beafc2e5758cd lib/Foswiki/Configure/Dependency.pm (Foswiki 1.1.4)
--- lib/Foswiki/Configure/Dependency.pm	2012-11-28 11:43:17.827220321 -0500
+++ lib/Foswiki/Configure/Dependency.pm.new	2012-11-28 11:40:57.174840122 -0500
@@ -459,6 +459,7 @@
         #print STDERR "Expecting SVN comparison, but RELEASE was $baseType \n";
         ( $baseType, @atuple ) = _decodeReleaseString($aVERSION)
           if ( defined $aVERSION && $baseType ne 'svn' );
+        return 1 if ( $baseType eq 'tuple' );
         return 0 unless ( $baseType eq 'svn' );
 
     }

~~~PATCH f446c7779d79cb9b9a59fa4cf62cd657:d02e6f96ebef1d1487af0d20333c827f lib/Foswiki/Configure/Dependency.pm (Foswiki 1.1.3)
--- lib/Foswiki/Configure/Dependency.pm	2012-11-28 11:46:00.510356565 -0500
+++ lib/Foswiki/Configure/Dependency.pm.new	2012-11-28 11:47:26.272377754 -0500
@@ -459,6 +459,7 @@
         #print STDERR "Expecting SVN comparison, but RELEASE was $baseType \n";
         ( $baseType, @atuple ) = _decodeReleaseString($aVERSION)
           if ( defined $aVERSION && $baseType ne 'svn' );
+        return 1 if ( $baseType eq 'tuple' );
         return 0 unless ( $baseType eq 'svn' );
 
     }

~~~PATCH 74546e2a4f1d06582d53b780a641c509:615f4a43b9ef72a732d2e4ac4cfa9981 lib/Foswiki/Configure/Dependency.pm (Foswiki 1.1.2)
--- lib/Foswiki/Configure/Dependency.pm	2012-11-28 11:48:54.014378721 -0500
+++ lib/Foswiki/Configure/Dependency.pm.new	2012-11-28 12:00:33.706477574 -0500
@@ -466,6 +466,7 @@
 
         # Didn't get a good RELEASE; fall back to subversion
         return 0 unless defined $aVERSION;
+        return 1 if ( $aVERSION =~ /^v?\d+\.\d+/ ); # a Dotted Decimal version string
         $aVERSION =~ s/^\s+//;
         $aVERSION =~ s/\s+$//;
         $aVERSION =~ s/^\$Rev: (\d+).*\$$/$1/;

~~~PATCH 0276c444cd7d5068434f7be5c88a7e6d:23d1bcea2820c6c25c4f1cf0d4bd8dda  lib/Foswiki/Configure/Dependency.pm (Foswiki 1.1.1)
--- lib/Foswiki/Configure/Dependency.pm	2012-11-28 12:35:38.838933114 -0500
+++ lib/Foswiki/Configure/Dependency.pm.new	2012-11-28 12:37:24.818753945 -0500
@@ -466,6 +466,7 @@
 
         # Didn't get a good RELEASE; fall back to subversion
         return 0 unless defined $aVERSION;
+        return 1 if ( $aVERSION =~ /^v?\d+\.\d+/ ); # a Dotted Decimal version string
         $aVERSION =~ s/^\s+//;
         $aVERSION =~ s/\s+$//;
         $aVERSION =~ s/^\$Rev: (\d+).*\$$/$1/;

~~~PATCH 62615300a753905267c1620321a1c51e:db9e7f7278873cc93d01c0607e18c17f lib/Foswiki/Configure/Dependency.pm (Foswiki 1.1.0)
--- lib/Foswiki/Configure/Dependency.pm	2012-11-28 12:38:58.889707506 -0500
+++ lib/Foswiki/Configure/Dependency.pm.new	2012-11-28 12:40:25.150748111 -0500
@@ -466,6 +466,7 @@
 
         # Didn't get a good RELEASE; fall back to subversion
         return 0 unless defined $aVERSION;
+        return 1 if ( $aVERSION =~ /^v?\d+\.\d+/ ); # a Dotted Decimal version string
         $aVERSION =~ s/^\s+//;
         $aVERSION =~ s/\s+$//;
         $aVERSION =~ s/^\$Rev: (\d+).*\$$/$1/;

